src/ewatercycle/esmvaltool/search.py

```diff
@@ -0,0 +1,219 @@
+    for name in valid_dataset_names:
+        ensembles = _get_value(
+            _filter_datasets(valid_datasets, "dataset", str(name)),
+            "ensemble",
+        )
+        valid_ensembles[str(name)] = set(str(ens) for ens in ensembles)
+    return valid_ensembles
+
+
+def _query_esgf(
+    activity: str,
+    experiment: str,
+    variables: list[str],
+    verbose: bool = False,
+) -> list[Dataset]:
+    """Return all datasets on ESGF that match the specified search query.
+
+    Args:
+        activity: Activity type, for example: 'ScenarioMIP'.
+        experiment: Experiment within the activity. E.g.: 'ssp585'
+        variables: Which variables are you searching for. Use the short_name definition.
+            For example: ['pr', 'tas'].
+        verbose (optional): If the results should be printed in a verbose way, to aid
+            in your search experience. Defaults to False.
+
+    Returns:
+        List of ESMValTool 'Dataset' objects that match the search query.
+    """
+    datasets = list()
+
+    for var in variables:
+        dataset_query = Dataset(
+            short_name=var,
+            activity=activity,
+            mip="*",
+            project="CMIP6",
+            exp=experiment,
+            dataset="*",
+            institute="*",
+            ensemble="*",
+            grid="*",
+        )
+        datasets_var = list(dataset_query.from_files())
+
+        if len(datasets_var) > 0 and verbose:
+            print(f"Found {len(datasets_var)} results for short name: {var}")
+            print("\n showing the first one.")
+            print(datasets_var[0])
+
+        datasets.extend(datasets_var)
+
+    return datasets
+
+
+def _get_mip_tables(
+    freq: Literal["hr", "3hr", "day"],
+    extended: bool = False,
+) -> tuple[str, ...]:
+    """Return the MIP tables that fit a certain desired data frequency.
+
+    Args:
+        freq: Desired frequency. 'hr', '3hr' or 'day'.
+        extended_mip_tables (optional): If you want to use extended MIP tables.
+            These tables are probably not relevant for most hydrology usecases and can
+            make the search slower. Defaults to False.
+
+    Returns:
+        The names of the MIP tables that match the desired frequency.
+    """
+    always_include: tuple[str, ...] = ("fx",)  # So orog is compatible with any freq.
+    mip_tables = {
+        "hr": ("E1hr",),
+        "3hr": ("3hr", "CF3hr", "E3hr"),
+        "day": ("day", "Eday", "CFday"),
+    }
+
+    extended_mip_tables = {
+        "hr": ("AERhr",),
+        "3hr": ("E3hrPt",),
+        "day": ("AERday", "Oday", "SIday"),
+    }
+    if freq not in mip_tables:
+        msg = (
+            f"Frequency '{freq}' is not implemented or not a valid mip table frequency"
+        )
+        raise ValueError(msg)
+
+    if extended:
+        return mip_tables[freq] + always_include + extended_mip_tables[freq]  # type: ignore
+    return mip_tables[freq] + always_include  # type: ignore
```

Comment:
When I run mypy v1.10.0 without the type ignore its still OK.  So the type ignore can be removed

```suggestion
        return mip_tables[freq] + always_include + extended_mip_tables[freq]
    return mip_tables[freq] + always_include
```

Reply:
I still had mypy 1.9.0 which complained about these lines. Good to know it's OK in 1.10!